DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/TestsForExtensions/GetPagesAmountExtensionTests.cs

```diff
@@ -0,0 +1,169 @@
+﻿using System.Collections.Generic;
+using FluentAssertions;
+using GitHubExtension.Activity.External.WebAPI.Exceptions;
+using GitHubExtension.Activity.External.WebAPI.Extensions;
+using GitHubExtension.Activity.External.WebAPI.Queries;
+using NSubstitute;
+using Xunit;
+
+namespace GitHubExtension.Activity.External.Tests.TestsForExtensions
+{
+    public class GetPagesAmountExtensionTests
+    {
+        public const string HeaderForShouldNotReturnNullWhenLinkHeaderIsNotNull =
+            "\"<https://api.github.com/repositories/52432402/events?page=2>; rel=\"next\", <https://api.github.com/repositories/52432402/events?page=10>; rel=\"last\"";
+
+        public const string HeaderForShouldNotThrowLinkHeaderMissingException =
+           "\"<https://api.github.com/repositories/52432402/events?page=2>; rel=\"next\", <https://api.github.com/repositories/52432402/events?page=10>; rel=\"last\"";
+
+        public const string HeaderForShouldNotThrowLinkHeaderFormatException =
+            "\"<https://api.github.com/repositories/52432402/events?page=2>; rel=\"next\", <https://api.github.com/repositories/52432402/events?page=10>; rel=\"last\"";
+
+        public const string HeaderForShouldThrowLinkHeaderFormatException =
+            "\"<https://api.github.com/repositories/52432402/events?page=2>; rel=\"next\", <https://api.github.com/repositories/52432402/events?page=10> rel=\"last\"";
+
+        public const string HeaderForLastRelNotPresent =
+            "\"<https://api.github.com/repositories/52432402/events?page=2>; rel=\"next\"";
+
+        public const string HeaderForShouldNotReturnNullWhenLastRelIsPresent =
+           "\"<https://api.github.com/repositories/52432402/events?page=2>; rel=\"next\", <https://api.github.com/repositories/52432402/events?page=10>; rel=\"last\"";
+
+        public const string HeaderForShouldThrowLinkHeaderFormatExceptionWhenLastRelPageIsNotNumber =
+            "\"<https://api.github.com/repositories/52432402/events?page=2>; rel=\"next\", <https://api.github.com/repositories/52432402/events?page=q10>; rel=\"last\"";
+
+        public const string HeaderForShouldParseLastRelHeaderCorrectly =
+            "\"<https://api.github.com/repositories/52432402/events?page=2>; rel=\"next\", <https://api.github.com/repositories/52432402/events?page=2>; rel=\"last\"";
+            
+        [Fact]
+        public void ShouldReturnNullWhenLinkHeaderIsNull()
+        {
+            // Arrange
+            IEnumerable<string> requestLinkHeader = null;
+            IGitHubEventsQuery query = Substitute.For<IGitHubEventsQuery>();
+
+            // Act
+            int? numberOfPages = query.GetNumberOfPages(requestLinkHeader);
```

Comment:
You are mocking IGitHubEventsQuery and then call GetNumberOfPages method? Are you sure?